### ISSUE TITLE
The PHPCS executable "/.config/composer/vendor/bin/phpcs" could not be found

### ISSUE THREAD

**Issue by jg314** [ObliviousHarmony/vscode-php-codesniffer#96]
### Description

When I load VS Code I get the following error in the Output panel: 'The PHPCS executable "/.config/composer/vendor/bin/phpcs" could not be found.' This errors continues to pop up many times as I make edits.

What's strange is that if I go into the settings and toggle any the options, it starts working without an issue, even if I don't change anything. For example, if I turn on and off the same checkbox, everything starts working as expected.

### Technical Setup
In case it's helpful, here are some details on my configuration:

Computer OS: Windows PC
VS Code Loaded In: Windows Subsystem for Linux (WSL2 running Ubuntu)
Composer Version: 2.5.8

I've also included a screenshot of some of the settings in VS Code below in case that's helpful.

![image](https://github.com/ObliviousHarmony/vscode-php-codesniffer/assets/1054379/1a1a022e-0aee-476a-883f-c742c1e900a7)

Thanks a ton.

**Comment by ObliviousHarmony** [comment]
That's very odd @jg314, especially that it is resolved by resetting the settings. I'll take a look at this and see if I can figure it out. Thanks for all of the additional information, it really helps!

**Comment by jg314** [comment]
Thanks a ton for taking a look @ObliviousHarmony. I really appreciate it.

**Comment by ObliviousHarmony** [comment]
I wasn't able to replicate it on macOS, I've got a few questions @jg314:

- Is your `phpcs` installed globally at `/.config/composer/vendor/bin/phpcs`?
- What's the path to your project directory?
- Does your project have `phpcs` installed?

I'll try it out on my Windows machine once I get a chance, but maybe the above can help me replicate it here.

**Comment by ObliviousHarmony** [comment]
Hi @jg314,

I tried this out on a Windows machine this morning and wasn't able to reproduce the issue. I used the same configuration that you gave above. Out of curiosity, how are you getting `phpcs` into your `$PATH`? I wonder if there's some reason that `phpcs` wouldn't be available on first run but after the settings were reset would become available. Could you share your `settings.json` configuration for `phpCodeSniffer` at the user, remote, and workspace levels?

If it isn't too much trouble, you could also clone the repo and do a little debugging! All you'd need to do is run `npm install` and then VS Code's `Run > Start Debugging` will take care of the rest. It will open a debug instance of VS Code that you can use to explore a project and use breakpoints and the like in the original editor. You could also just add `console.log()` calls! In particular, it would be useful to see what `Worker::createProcess()` has for `executable` and `processArguments`.

**Comment by jg314** [comment]
I think I may have found the issue. I went through all the settings related to the extension and found this in my user settings JSON:

`"phpCodeSniffer.executable": "/.config/composer/vendor/bin/phpcs",`

That path is incorrect and when I remove it, everything starts working correctly and the errors goes away. What's strange is that there's no setting for this within the settings UI. I also doubt I would have set that manually since it's the wrong location for the executable.

In any case, I'm hoping this resolves the issue for good. Thanks a ton for all your help troubleshooting this. I really appreciate the time and the work you did to make the extension available inside VS Code.

**Comment by ObliviousHarmony** [comment]
I figured that might be it @jg314 :smile: When I tested this on my Windows machine I ran into the _exact same problem_ from an old configuration I had in place to test Docker executables out. I'm not sure why it doesn't show up in the settings but that's probably an upstream problem in VS Code. It's _supposed_ to display a setting as deprecated when `markdownDeprecationMessage` is set, and it used to work, but it doesn't seem to anymore.
